vsintegration/src/FSharp.Editor/LanguageService/LanguageService.fs

```diff
@@ -417,49 +479,110 @@ and
 
     member this.SetupProjectFile(siteProvider: IProvideProjectSite, workspace: VisualStudioWorkspaceImpl, userOpName) =
         let userOpName = userOpName + ".SetupProjectFile"
-        let  rec setup (site: IProjectSite) =
+        let rec setup (site: IProjectSite) =
             let projectGuid = Guid(site.ProjectGuid)
             let projectFileName = site.ProjectFileName()
             let projectDisplayName = projectDisplayNameOf projectFileName
-
             let projectId = workspace.ProjectTracker.GetOrCreateProjectIdForPath(projectFileName, projectDisplayName)
+            projectInfoManager.UpdateProjectInfo(tryGetOrCreateProjectId workspace, projectId, site, workspace, userOpName)
 
-            if isNull (workspace.ProjectTracker.GetProject projectId) then
```

Comment:
Previously we used this line to guard against doing work multiple times for the same project as we traverse the project graph

You've moved the check down a bit, but this means that ``UpdateProjectInfo`` may now be called many times per project, along with actions such as creating ProjectExternalErrorReporter, and, most critically, the recursive calls and AddProjectReference calls here:

```
            for referencedSite in ProjectSitesAndFiles.GetReferencedProjectSites (site, this.SystemServiceProvider) do
                let referencedProjectId = setup referencedSite
                project.AddProjectReference(ProjectReference referencedProjectId)
```

I think you will want to somehow want to guard against doing these things many times as we traverse the project graph.